packages/ui-library/src/components/forms/number-input/index.css.ts

```diff
@@ -1,181 +1,173 @@
+        gap: ${NumberInput.Input.TextContainer.ItemSpacing.SM};
+      }
 
-        &.lg {
-          > .custom-stepper-button {
-            width: ${StepperCombo.LG.Vertical.Width};
-          }
-        }
-        > .unit {
-          text-align: left;
-        }
-  
-        > input {
-          text-align: right;
-          &.sm{
-            padding-right:${NumberInput.Input.TextContainer.ItemSpacing.SM};
-          }
+      &.md {
+        gap: ${NumberInput.Input.TextContainer.ItemSpacing.MD};
+      }
 
-          &.md{
-            padding-right:${NumberInput.Input.TextContainer.ItemSpacing.MD};
-          }
+      &.lg {
+        gap: ${NumberInput.Input.TextContainer.ItemSpacing.LG};
+      }
 
-          &.lg{
-            padding-right:${NumberInput.Input.TextContainer.ItemSpacing.LG};
-          }
-        }
+      .unit {
+        color: ${PrefixSuffix.OnPopulatedField.Default.Rest};
       }
+    }
 
-      > .unit,
-      input {
-        &.sm {
-          font-weight: ${SM.UserInput.fontWeight};
-          font-size: ${SM.UserInput.fontSize};
-          font-family: ${SM.UserInput.fontFamily}, sans-serif;
-          line-height: ${SM.UserInput.lineHeight};
-          padding: ${SM.InputField.Padding};
+    input {
+      border: none;
+      outline: none;
+    }
 
-        }
+    & > * {
+      background-color: transparent;
+    }
 
-        &.md {
-          font-weight: ${MD.UserInput.fontWeight};
-          font-size: ${MD.UserInput.fontSize};
-          font-family: ${MD.UserInput.fontFamily}, sans-serif;
-          line-height: ${MD.UserInput.lineHeight};
-          padding: ${MD.InputField.Padding};
+    .split {
+      &.sm {
+        & > .custom-stepper-button {
+          width: ${StepperCombo.SM.Vertical.Width};
         }
+      }
 
-        &.lg {
-          font-weight: ${LG.UserInput.fontWeight};
-          font-size: ${LG.UserInput.fontSize};
-          font-family: ${LG.UserInput.fontFamily}, sans-serif;
-          line-height: ${LG.UserInput.lineHeight};
-          padding: ${LG.InputField.Padding};
+      &.md {
+        & > .custom-stepper-button {
+          width: ${StepperCombo.MD.Vertical.Width};
         }
-
       }
 
-      &.disabled {
-        outline: ${Input.Default.Disabled.width} ${Input.Default.Disabled.style} ${Input.Default.Disabled.color};
-        background-color: ${SurfaceFill.Default.Disabled};
-        cursor: not-allowed;
-
-        & > input {
-          color: ${UserInput.Default.Disabled};
-          cursor: not-allowed;
-
-          &::placeholder {
-            color: ${Placeholder.Default.Disabled};
-          }
+      &.lg {
+        & > .custom-stepper-button {
+          width: ${StepperCombo.LG.Vertical.Width};
         }
       }
+    }
 
-      &:focus-within {
-        outline: ${Input.Default.Focus.width} ${Input.Default.Focus.style} ${Input.Default.Focus.color};
-        background-color: ${SurfaceFill.Default.Focus};
+    .unit,
+    input {
+      &.sm {
```

Comment:
We may want to consider making the width of the input 100% for all sizes. I noticed that we can't click (or focus) on the input for most of the space it fills because the width is small. It may look ok visually, but it might be a better user experience if they have more space to click on the input. Hopefully the screenshot below helps clarify what I mean
<img width="1075" alt="Screenshot 2024-01-22 at 14 12 02" src="https://github.com/deven-org/B01LER-Kitchen/assets/127498135/ab21937a-4e32-440c-b375-d9dcc9cda650">

Reply:
Making this change will also fix the readonly version:
Before:
<img width="952" alt="Screenshot 2024-01-22 at 14 16 15" src="https://github.com/deven-org/B01LER-Kitchen/assets/127498135/31393cbe-de93-4950-ba39-2090b064cc5c">
After:
<img width="945" alt="Screenshot 2024-01-22 at 14 16 20" src="https://github.com/deven-org/B01LER-Kitchen/assets/127498135/dba7ad4a-8881-41e8-ba9f-42b8d695fca1">

Reply:
To the first topic: if the input becomes full width, we can not center it in split mode like seen in figma. its actually the opposed thing we need here.... it needs to shrink/grow depending on its content. the entire wrapper would become the right cursors, and no matter where you click (except buttons) it would gain focus.

This was discussed to be done in a follow up ticket. 

but the read only paddings needs to be adressed